File: Sliding Window Pattern/Maximum_Sum_Of_Any_Contiguous_Subarray.py
def contiguousSubarrayOfSizeKWithMaxSum(array, k): # O(n) Time Complexity
    max_sum = 0
    window_sum = 0
    window_start = 0
    for window_end in range(len(array)):
        window_sum = window_sum + array[window_end]
        if window_end >= k - 1:
            max_sum = max(window_sum, max_sum)
            window_sum = window_sum - array[window_start]
            window_start = window_start + 1
    return max_sum

if __name__ == "__main__":
    array = [2, 1, 5, 1, 3, 2]
    k = 3
    result = contiguousSubarrayOfSizeKWithMaxSum(array, k)
    print(result)

File: Sliding Window Pattern/Maximum_Sum_Of_Any_Contiguous_Subarray.cpp
#include <iostream>
#include <vector>
using namespace std;
 
int contiguousSubarrayOfSizeKWithMaxSum(vector<int> array, int k) {
    int windowStart = 0, maxSum = 0, windowSum = 0;
    for (int windowEnd = 0; windowEnd < array.size(); windowEnd++) {
        windowSum = windowSum + array[windowEnd];
        if (windowEnd >= (k - 1) ) {
            maxSum = max(maxSum, windowSum);
            windowSum = windowSum - array[windowStart];
            windowStart = windowStart + 1;
        }
    }
    return maxSum;
}

int main() {
    vector<int> array = {2, 1, 5, 1, 3, 2};
    int k = 3;
    int reuslt = contiguousSubarrayOfSizeKWithMaxSum(array, k);
    cout<<reuslt<<endl;
    return 0;
}